developer/wip/for-contributors/control-plane/protocol-extensions.md

```diff
@@ -0,0 +1,94 @@
+# Protocol Extensions
+
+<!-- TOC -->
+* [Protocol Extensions](#protocol-extensions)
+  * [1. RemoteMessage](#1-remotemessage)
+    * [1.1 Delivering messages with RemoteMessageDispatcher](#11-delivering-messages-with-remotemessagedispatcher)   
+    * [1.2 Handling incoming messages with protocol services](#12-handling-incoming-messages-with-protocol-services)   
+  * [2. DSP protocol](#2-dsp-protocol-implementation)
+    * [2.1 `RemoteMessage` handlers](#21-remotemessage-handlers)
+    * [2.2 HTTP endpoints](#22-http-endpoints)
+    * [2.3 `RemoteMessage` transformers](#22-remotemessage-transformers)
+
+<!-- TOC -->
+
+
+The EDC officially supports the [Dataspace protocol](#2-dsp-protocol) using the HTTPs bindings, but since it is an extensible platform, multiple protocol implementations can be supported for inter-connectors communication. Each supported protocols is identified by a unique key used by EDC for dispatching a [remote message](#1-remotemessage). 
+
+## 1. RemoteMessage
+
+At the heart of EDC message exchange mechanism lies the `RemoteMessage` interface, which describes the `protocol`, the `counterPartyAddress` and the `counterPartyId` used for a message delivery. 
+
+`RemoteMessage` extensions can be divided in three groups:
+
+- [Catalog](./entities.md#6-catalog) (`catalog-spi`)
+- [Contract negotiation](./entities.md#4-contract-negotiations) (`contract-spi`)
+- [Transfer process](./entities.md#7-transfer-processes) (`transfer-spi`)
+
+Each `RemoteMessage` is:
+
+- Delivered to the counter-party using a [RemoteMessageDispatcher](#11-delivering-messages-with-remotemessagedispatcher). 
+- Received by the counter-party through the protocol implementation (e.g HTTP) and handled by the [protocol services](#11-handling-ingress-messages-with-protocol-services) layer.
+
+
+### 1.1 Delivering messages with RemoteMessageDispatcher
+
+Each protocol implements a `RemoteMessageDispatcher`:
+
+```java
+public interface RemoteMessageDispatcher {
+
+    String protocol();
+
+    <T, M extends RemoteMessage> CompletableFuture<StatusResult<T>> dispatch(Class<T> responseType, M message);
+
+}
+```
+and it is registered in the `RemoteMessageDispatcherRegistry`, where it gets associated to the protocol defined in `RemoteMessageDispatcher#protocol`.
+
+Internally EDC uses the `RemoteMessageDispatcherRegistry` whenever it needs to deliver a `RemoteMessage` to the counter-party. The `RemoteMessage` then gets routed to the right `RemoteMessageDispatcher` based on the `RemoteMessage#getProtocol` property.
```

Comment:
it might be worth mentioning, that we are using the message dispatcher mechanism also for non-protocol messages, e.g. event callbacks (Webhooks)